Hmac.js

```diff
@@ -57,20 +58,37 @@ export class Hmac {
    * @returns {Promise<string>} The base64url-encoded signature.
    */
   async sign({data, useCache = true}) {
-    const cacheKey = `sign-${base64url.encode(data)}`
+    const cacheKey = `sign-${base64url.encode(data)}`;
     if(useCache) {
       const signature = this.cache.get(cacheKey);
       if(signature) {
         return signature;
       }
     }
-    const {id: keyId, kmsClient, capability, invocationSigner} = this;
-    const signature = await kmsClient.sign(
```

Comment:
Why not get the promise that results from `kmsClient.sign` and put it in the cache ... then try/catch await it and remove it from the cache if you fall into catch?